File: ruby/2.0.0/mysql/mysql.rb
require 'mysql2'


begin
    connection = Mysql2::Client.new(:host => "localhost", :username => "test", :password => "test", :database => 'test')
    puts connection.server_info
    resultSet = connection.query 'SELECT * from t'
    resultSet.each do |row|
      puts 'Data row = (%s, %s, %s)' % [row['id'], row['name'], row['quantity']]
    end
    puts 'Read ' + resultSet.count.to_s + ' row(s).'
    
rescue Mysql2::Error => e
    puts e.errno
    puts e.error
    
ensure
    connection.close if connection
end

File: node.js/4.2.4/mysql/mysql.js
// ALTER USER 'test'@'localhost' IDENTIFIED WITH mysql_native_password BY 'test';
var mysql      = require('mysql');
var connection = mysql.createConnection({
  host     : 'localhost',
  user     : 'test',
  password : 'test',
  database : 'test'
});

connection.connect();

connection.query('SELECT * from t', function(err, rows, fields) {
  if (!err)
    console.log('The solution is: ', rows);
  else
    console.log('Error while performing Query: ', err);
});

connection.end();
